presto-main/src/main/java/com/facebook/presto/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -199,14 +206,23 @@ public void setOutputBuffers(OutputBuffers newOutputBuffers)
     }
 
     @Override
-    public void enqueue(List<SerializedPage> pages)
+    public void registerLifespanCompletionCallback(Consumer<Lifespan> callback)
+    {
+        checkState(lifespanCompletionCallback == null, "lifespanCompletionCallback is already set");
+        this.lifespanCompletionCallback = requireNonNull(callback, "callback is null");
+    }
+
+    @Override
+    public void enqueue(Lifespan lifespan, List<SerializedPage> pages)
     {
         checkState(!Thread.holdsLock(this), "Can not enqueue pages while holding a lock on this");
+        requireNonNull(lifespan, "lifespan is null");
         requireNonNull(pages, "page is null");
+        checkState(lifespanCompletionCallback != null, "lifespanCompletionCallback has to be set before starting to enqueue data");
```

Comment:
`lifespanCompletionCallback is not set`
OR
`lifespanCompletionCallback must be set before enqueueing data`

Reply:
Also fixed for other two buffer :)